pkg/cryptoinfo/info.go

```diff
@@ -13,18 +13,52 @@ type KeyInfo struct {
 
 // Maybe make these types?
 const (
-	kiPEM         = "PEM"
-	kiDER         = "DER"
-	kiCertificate = "CERTIFICATE"
+	kiPEM = "PEM"
+	kiDER = "DER"
+	kiP12 = "P12"
+
+	kiCACERTIFICATE = "CA-CERTIFICATE" // What is correct here?
+	kiCaCertificate = "certificate"
+	kiCERTIFICATE   = "CERTIFICATE"
+	kiCertificate   = "certificate"
+	kiKEY           = "KEY"
+	kiKey           = "key"
 )
 
-func NewKeyInfo(typ, encoding string, headers map[string]string) *KeyInfo {
+func NewKIKey(encoding string) *KeyInfo {
```

Comment:
May I suggest a naming scheme like: `NewKeyKeyInfo/NewCertificateKeyInfo` etc?

Or, we could probably forgo the Hungarian-ish notation, since the return types of the funcs are probably documentation enough. Then we could just name these `NewKey` `NewCertificate`, `NewCaCertificate`, `NewError` etc. 

I think I like the latter option better actually..

Reply:
Given the package scope, I think your latter suggestion is pretty good. Yeah.